### PR TITLE
[Refactor] Remove old partition level tablet type

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Problem Summary(Required) ：

1. We will introduce StarRocks LakeTable, so remove partition level
useStarOS.
2. Remove S3 TStorageMedium that has been not used.
3. Remove old UTs and add news with LakeTable.

### CODE REVIEW

**Review by sduzh — DISMISSED** [review]

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 6 / 10 (60.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndex.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/TabletMeta.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletMeta.java [hunk 1]
```diff
@@ -82,8 +82,9 @@ public void setStorageMedium(TStorageMedium storageMedium) {
         this.storageMedium = storageMedium;
     }
 
+    // Remove this after LakeTable is merged.
```
Comment (wuleistarrocks) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TabletMeta.java:85]: TODO is suggested

File: fe/fe-core/src/main/java/com/starrocks/catalog/Partition.java [hunk 2]
```diff
@@ -200,12 +195,9 @@ public DistributionInfo getDistributionInfo() {
         return distributionInfo;
     }
 
-    public void setPartitionInfo(PartitionInfo partitionInfo) {
-        this.partitionInfo = partitionInfo;
-    }
-
+    // Remove this after LakeTable is merged.
```
Comment (wuleistarrocks) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Partition.java:198]: TODO is suggested

File: fe/fe-core/src/main/java/com/starrocks/catalog/TabletMeta.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TabletMeta.java]: ok

**Review by wuleistarrocks — APPROVED** [review]

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Review by abc982627271 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 6 / 10 (60.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/proc/IndicesProcDir.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Partition.java|3|4|75.00%|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/MaterializedIndex.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/catalog/TabletMeta.java|1|1|100.00%|

**Review by sduzh — APPROVED** [review]

**Pull request merged**